text/0088-metric-instrument-optional-refinements.md

```diff
@@ -0,0 +1,483 @@
+properties, it is natural to combine the distributions.  The
+distinction is about how we interpret the values when aggregating.
+Use one of the sum-only refinments to report a sum in the default
+configuration, otherwise use one of the non-sum-only instruments to
+report a distribution.
+
+#### Language-level refinements
+
+OpenTelemetry implementations may wish to add instrument refinements
+to accommodate built-in types.  Languages with distinct integer and
+floating point should offer instrument refinements for each, leading
+to type names like `Int64Measure` and `Float64Measure`.
+
+A language with support for unsigned integer types may wish to create
+dedicated instruments to report these values, leading to type names
+like `UnsignedInt64Observer` and `UnsignedFloat64Observer`.  These
+would naturally apply a non-negative refinment.
+
+Other uses for built-in type refinements involve the type for duration
+measurements.  For example, where there is built-in type for the
+difference between two clock measurements, OpenTelemetry APIs should
+offer a refinement to automatically apply the correct unit of time to
+the measurement.
+
+### Counter refinement
+
+Counter is a sum-only, non-negative, thus non-negative-rate refinement
+of the Measure instrument.
+
+### Standardizing new instruments
+
+With these refinements we can exhaustively list each distinct kind of
+instrument.  There are a total of twelve hypothetical instruments
+listed in the table below, of which only one has been standardized.
+Hypothetical future instrument names are _italicized_.
+
+| Foundation instrument | Sum-only? | Precomputed-sum? | Non-negative? | Non-negative-rate? | Instrument name _(hyptothetical)_ |
+|--|--|--|--|--|--|
+| Measure  | sum-only |                 | non-negative  | non-negative-rate | Counter |
+| Measure  | sum-only | precomputed-sum |               | non-negative-rate | _CumulativeCounter_ |
+| Measure  | sum-only |                 |               |                   | _UpDownCounter_ |
+| Measure  | sum-only | precomputed-sum |               |                   | _UpDownCumulativeCounter_ |
+| Measure  |          |                 | non-negative  |                   | _AbsoluteMeasure_ |
+| Measure  |          |                 |               |                   | _NonAbsoluteMeasure_ |
+| Observer | sum-only |                 | non-negative  | non-negative-rate | _DeltaObserver_ |
+| Observer | sum-only | precomputed-sum |               | non-negative-rate | _CumulativeObserver_ |
+| Observer | sum-only |                 |               |                   | _UpDownDeltaObserver_ |
+| Observer | sum-only | precomputed-sum |               |                   | _UpDownCumulativeObserver_ |
+| Observer |          |                 | non-negative  |                   | _AbsoluteObserver_ |
+| Observer |          |                 |               |                   | _NonAbsoluteObserver_ |
```

Comment:
This needs to be updated based on https://github.com/open-telemetry/oteps/pull/88#discussion_r398204203

Reply:
Done. (Thanks for pushing this resolution.)